transcription/services.py

```diff
@@ -895,6 +895,76 @@ def generate_session_log_from_transcripts(
         gamelog.save(update_fields=["generated_log_text"])
         return session_log
 
+    def process_session_audio_async(
+        self,
+        session_audio: SessionAudio,
+        previous_transcript: str = "",
+        session_notes: str = "",
+        use_celery: bool = True,
+    ):
+        """
+        Process a SessionAudio instance asynchronously using Celery.
+        
+        Args:
+            session_audio: The SessionAudio instance to process
+            previous_transcript: Previous transcript text for context
+            session_notes: Session notes for context
+            use_celery: Whether to use Celery for async processing
+            
+        Returns:
+            AsyncResult if using Celery, otherwise result of synchronous processing
+        """
+        if use_celery:
+            try:
+                from .tasks import process_session_audio_task
+                return process_session_audio_task.delay(
+                    session_audio.id, previous_transcript, session_notes
+                )
+            except ImportError:
+                # Fallback to synchronous processing if Celery is not available
+                return self.process_session_audio(
```

Comment:
I know synchronous processing will fail, or at least it does from a django admin action. If it's sure to fail, I'd rather give a helpful error saying that celery isn't running or something, if that's a failpoint.